docs/source/upgrading/4.0-4.1.rst

```diff
@@ -20,148 +20,7 @@ SSL enabled clients and servers.
-Pay attention to the :doc:`security considerations </experimental/httpserver-ssl>` and limitations using this on a microcontroller with limited RAM memory::
-
-  void startWebServer()
-  {
-  #ifdef ENABLE_SSL
-    server.setSslInitHandler([](Ssl::Session& session) {
-      debug_i("SSL Init handler: setting server keyCert");
-      session.keyCert.assign(serverKey, serverCert);
-    });
-    server.listen(443, true);
-  #else
-    server.listen(80);
-  #endif
-    server.paths.set("/", onIndex);
-    server.paths.set("/ipconfig", onIpConfig);
-    server.paths.set("/ajax/get-networks", onAjaxNetworkList);
-    server.paths.set("/ajax/connect", onAjaxConnect);
-    server.paths.setDefault(onFile);
-  }
-
-
-Setting client certificates, ssl options and pinning for a HttpRequest is done using onSslInit callback.
-If you look at the :sample:`Basic_Ssl` sample you will see that the old way of setting them was as shown below::
-
-  HttpRequest* request = new HttpRequest(F("https://www.grc.com/fingerprints.htm"));
-  request->setSslOptions(SSL_SERVER_VERIFY_LATER);
-  request->pinCertificate(fingerprints);
-
-The new one is using the following sequence of commands::
-
-  auto request = new HttpRequest(F("https://www.grc.com/fingerprints.htm"));
-  request->onSslInit(grcSslInit);
-
-
-A sample `grcSslInit` callback is given below. In the callback the developer
-has access to the current SSL session and HTTP request and can modify them accordingly::
-
-  void grcSslInit(Ssl::Session& session, HttpRequest& request)
-  {
-    SslFingerprints fingerprints;
-
-    //...
-
-    session.validators.add(fingerprints);
-
-    // We're using fingerprints, so don't attempt to validate full certificate
-    session.options.verifyLater = true;
-
-    session.fragmentSize = Ssl::eSEFS_16K;
-  }
-
-
-Cryptographic functions
-~~~~~~~~~~~~~~~~~~~~~~~
-Cryptographic functions are now not included by default. This means that if you want to continue using them they should be included
-in your application source code. For example the following old code is using axTLS cryptographic functions::
-
-
-  char* loadPsk(int* keylen)
-  {
-    SHA1_CTX sha_ctx;
-    // ...
-    SHA1_Init(&sha_ctx);
-    SHA1_Update(&sha_ctx, (uint8_t*)buffer, strlen(buffer));
-    SHA1_Final(digest, &sha_ctx);
-
-For this code to work you should include the following header::
-
-  #include <axtls-8266/crypto/crypto.h>
-
-
-And also make sure that your application `component.mk` file has the following line::
-
-  COMPONENT_DEPENDS += axtls-8266
-
-You can use also one SSL adapter, for example Bearssl, with another cryptographic library, for example Axtls.
-
-SSL namespace
-~~~~~~~~~~~~~
-All SSL related classes and types are organized in a separate namespace called `Ssl`. For example you should use `Ssl::KeyCertPair`
-instead of `SslKeyCertPair` and `Ssl::Fingerprints` instead of `SslFingerprints`.
+See :component:`ssl` for further details.
```

Comment:
Can you hint the user to go directly to the SSL upgrade document ( https://sming.readthedocs.io/en/feature-crypto/_inc/Sming/Components/ssl/upgrade.html )?